### PR TITLE
Unable to login through Settings | Tools | Azure | Service Authentication 

### LINKED ISSUE #518: Unable to login through Settings | Tools | Azure | Service Authentication
JetBrains Rider 2021.2 RD-212.4746.114
Azure Toolkit for Rider 3.50.0.1253-2021.2

**STR:**
1. Open `File | Settings | Tools | Azure | Service Authentication`
2. Click "Sign In"
3. Choose "Azure CLI" or "Device Code" and click "Sign In"

**Actual:**
Nothing happens.

### PR DESCRIPTION
**Pull request by maartenba**
Fixes #518 9and an annoying focus stealing issue)
Relates to #514

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/SubscriptionsDialog.java [hunk 1]
```diff
@@ -138,10 +139,25 @@ private void createUIComponents() {
...
         JTableUtils.enableBatchSelection(table, CHECKBOX_COLUMN);
         table.getTableHeader().setReorderingAllowed(false);
         new TableSpeedSearch(table);
 
+        // Workaround for focus stealing after clicking in the checkbox column.
+        // Observed while testing https://github.com/JetBrains/azure-tools-for-intellij/issues/518
+        table.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(final FocusEvent e) {
+                if (isModal() &&
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/SubscriptionsDialog.java:152]: Does it mean this modal window is on the top unless we select a subscription?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/identity/managed/AzureManagedIdentityConfigurationPanel.kt [hunk 2]
```diff
@@ -61,11 +62,11 @@ class AzureManagedIdentityConfigurationPanel(private val project: Project) : Azu
             lateinit var signInRow: Row
             signInRow = row {
                 button(RiderAzureBundle.message("settings.managedidentity.sign_in")) {
-                    AuthMethodManager.getInstance().signOut()
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/identity/managed/AzureManagedIdentityConfigurationPanel.kt:64]: Why do we have it here? No side effects after removing this line?

Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/org/jetbrains/plugins/azure/identity/managed/AzureManagedIdentityConfigurationPanel.kt:64]: The `doSignIn` action handles this internally.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/SubscriptionsDialog.java (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/SubscriptionsDialog.java:152]: It steals focus after (de)selecting a subscription. Clicking a button in the dialog is fine.

**Review by sdubov — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/SubscriptionsDialog.java (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/SubscriptionsDialog.java:152]: For future reference: the window itself is modal already - https://github.com/JetBrains/azure-tools-for-intellij/pull/519/files#diff-933855ad3b71863bc11dab5c7b079c786f7439a4b5a4a6943782ac86c5259079R57

**Pull request merged**